### PR TITLE
Reduced the memory consumption in the event based risk calculator

### PR DESCRIPTION
**Pull request by micheles**
Aggregating by array instead that by list of arrays. In our demo the improvement is from ~900 KB per task to ~300 KB per task (pickled size of the task result). Moreover, I am reducing the data transfer back by removing big attributes from the monitor. The difference is from ~73 MB to ~300 KB per task.
Closes https://github.com/gem/oq-risklib/issues/451

### CODE REVIEW

File: openquake/calculators/event_based_risk.py [hunk 1]
```diff
@@ -71,10 +78,13 @@ def event_based_risk(riskinputs, riskmodel, rlzs_assoc, monitor):
         a single array of dtype elt_dt, or an empty list
     """
     lti = riskmodel.lti  # loss type -> index
-    specific_assets = monitor.oqparam.specific_assets
-    result = cube(
-        monitor.num_outputs, len(lti), len(rlzs_assoc.realizations), list)
-    for out_by_rlz in riskmodel.gen_outputs(riskinputs, rlzs_assoc, monitor):
+    L, R = len(lti), len(rlzs_assoc.realizations)
```
Comment (nastasi-oq) [inline openquake/calculators/event_based_risk.py]: L, R, l, r, some more letter in var names, please.

Comment (micheles) [inline openquake/calculators/event_based_risk.py]: There is a long tradition in scientific computing about using one-letter variable for integers. Also, the names are used in all other calculators, so I want to keep them. `r` is the loop index in `range(R)`, `l` is the loop index in `range(L)`.

Comment (nastasi-oq) [inline openquake/calculators/event_based_risk.py]: Few understandable in any case.
Maybe a document with this kind of practices could help developers in the future.

**Comment by nastasi-oq** [comment]
LGTM

**Pull request merged**